### PR TITLE
Added ability to list and create labels

### PR DESCRIPTION
**Pull request by jjimenezlopez**
Hi,

I added the ability to list and create project labels. This is my first time using coffee script, so feel free to reject the pull-request or tell me if I have to do something in a different way.

Regards!

### CODE REVIEW

**Comment by moul** [comment]
Looks good to me :ok_hand: 

@dave-irvine is it ok for you too ?

File: src/Models/Labels.coffee [hunk 1]
```diff
@@ -0,0 +1,14 @@
+BaseModel = require '../BaseModel'
+
+class Labels extends BaseModel
+
+  create: (projectId, params = {}, fn = null) =>
+    @debug "Labels::create()"
+    if projectId.toString().indexOf("/") isnt -1
```
Comment (dave-irvine) [inline src/Models/Labels.coffee]: Doesn't `Utils.parseProjectId` handle this?

File: src/Models/ProjectLabels.coffee [hunk 2]
```diff
@@ -0,0 +1,31 @@
+BaseModel = require '../BaseModel'
+Utils = require '../Utils'
+
+class ProjectLabels extends BaseModel
+  list: (projectId, params = {}, fn = null) =>
```
Comment (dave-irvine) [inline src/Models/ProjectLabels.coffee]: I think following the conventions we have set elsewhere, we'd call this `all()`.

File: src/Models/ProjectLabels.coffee [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    @debug "ProjectLabels::labels()"
+
+    if 'function' is typeof params
+      fn = params
+      params = {}
+    params.page ?= 1
+    params.per_page ?= 100
+
+    do (->
```
Comment (dave-irvine) [inline src/Models/ProjectLabels.coffee]: Check out the way I've done this in Projects.coffee:

https://github.com/node-gitlab/node-gitlab/blob/develop/src/Models/Projects.coffee#L24

I originally wrote it the way you've done here (I assume you copy/pasted), but figured out a cleaner way a month or so back.

File: src/Models/ProjectLabels.coffee (hunk 2 above)
Comment (jjimenezlopez) [inline src/Models/ProjectLabels.coffee]: In *ProjectIssues* this method is also called `list()`, should I change this to `all()`?

File: src/Models/ProjectLabels.coffee (hunk 3 above)
Comment (jjimenezlopez) [inline src/Models/ProjectLabels.coffee]: Yes, I copied it, I'll take a look later today.

**Comment by jjimenezlopez** [comment]
Ok, I'll take a look later today and I'll add the commits to this pull-request.

File: src/Models/ProjectLabels.coffee (hunk 2 above)
Comment (dave-irvine) [inline src/Models/ProjectLabels.coffee]: Yes we need to change ProjectIssues but we should do that in a seperate pull request.

**Comment by dave-irvine** [comment]
Don't forget to add yourself to the README under Contributors.

File: src/Models/ProjectLabels.coffee (hunk 3 above)
Comment (dave-irvine) [inline src/Models/ProjectLabels.coffee]: Cool. The change is easy, here it is:

https://github.com/node-gitlab/node-gitlab/commit/698d5d7970fae9f9b1e4522acd029f3662b965aa

**Comment by jjimenezlopez** [comment]
Done : )

**Comment by dave-irvine** [comment]
Great, merged :) :+1:

**Comment by moul** [comment]
:+1: thanks to both of you

**Pull request merged**